java/connector-node/risingwave-sink-mock-flink/risingwave-sink-mock-flink-common/src/main/java/com/risingwave/connector/common/FlinkDynamicUtil.java

```diff
@@ -0,0 +1,198 @@
+                                        flinkColumnMap.get(columnDesc.getName()),
+                                        columnDesc.getDataType().getTypeName()))
+                        .asRuntimeException();
+            }
+        }
+    }
+
+    private static List<Column> getColumns(
+            TableSchema tableSchema,
+            FlinkDynamicAdapterConfig config,
+            CatalogFactory catalogFactory) {
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+        Configuration configuration = new Configuration();
+        Set<ConfigOption<?>> configOptions = catalogFactory.requiredOptions();
+        configOptions.addAll(catalogFactory.optionalOptions());
+        config.processOption(configOptions);
+
+        Schema.Builder schemaBuilder = Schema.newBuilder();
+        tableSchema
+                .getColumnDescs()
+                .forEach(
+                        (columnDesc) ->
+                                schemaBuilder.column(
+                                        columnDesc.getName(),
+                                        getCorrespondingFlinkType(columnDesc.getDataType())));
+        // Start with the default value, and add as needed later
+        FactoryUtil.DefaultCatalogContext defaultCatalogContext =
+                new FactoryUtil.DefaultCatalogContext(
+                        "catalog", config.getOption(), configuration, contextClassLoader);
+        Catalog catalog = catalogFactory.createCatalog(defaultCatalogContext);
+
+        List<Column> columns = new ArrayList<>();
+        try {
+            catalog.open();
+            List<Schema.UnresolvedColumn> flinkColumns =
+                    catalog.getTable(config.getTablePath()).getUnresolvedSchema().getColumns();
+            catalog.close();
+            for (int i = 0; i < flinkColumns.size(); i++) {
+                Schema.UnresolvedColumn unresolvedColumn = flinkColumns.get(i);
+                if (unresolvedColumn instanceof Schema.UnresolvedPhysicalColumn) {
+                    Schema.UnresolvedPhysicalColumn c1 =
+                            (Schema.UnresolvedPhysicalColumn) unresolvedColumn;
+                    columns.add(Column.physical(c1.getName(), (DataType) (c1.getDataType())));
+                } else {
+                    throw new RuntimeException("Only support physical column");
+                }
+            }
+            return columns;
+        } catch (TableNotExistException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static boolean checkType(Data.DataType dataType, DataType flinkType) {
+        switch (dataType.getTypeName()) {
+            case DECIMAL:
+                return flinkType.toString().contains("DECIMAL");
+            case TIMESTAMP:
+                return flinkType.toString().contains("TIMESTAMP");
+            case TIMESTAMPTZ:
+                String pattern = "TIMESTAMP\\(\\d+\\) WITH TIME ZONE";
```

Comment:
nits: the regex can be a static variable so that we don't have to recreate the object for multiple times.